ecs-cli/modules/cli/compose/project/project.go

```diff
@@ -130,6 +136,21 @@ func (p *ecsProject) parseCompose() error {
 	return p.context.SetProjectName()
 }
 
+// parseECSParams sets data from the ecs-params.yml file on the ecsProject.context
+func (p *ecsProject) parseECSParams() error {
+	logrus.Debug("Parsing the ecs-params yaml...")
+	ecsParamsFileName := p.context.CLIContext.GlobalString(command.ECSParamsFileNameFlag)
```

Comment:
Be careful about `context.GlobalString`, it doesn't actually do what its name implies it does. So in urfave/cli, each subcommand gets its own context. So in `ecs-cli compose service up` the lowest level context is flags for `up`, then its parent has flags for `service` and so on. `GlobalString` simply looks in the parent context. As a result of this, I wrote this function to recursively search up the context: https://github.com/aws/amazon-ecs-cli/blob/master/ecs-cli/modules/config/params.go#L37

`GlobalString` should actually be named `ParentString` in my opinion.

I don't know if this will actually cause a bug in your code. But it might be worthwhile to move the recursive function to a different package, export it, and use it here (and elsewhere) in the CLI.